### ISSUE TITLE
Incorrect work with coordinates in landscape mode while executing press + moveTo commands

### ISSUE THREAD

**Issue by tmytlovych** [facebook/WebDriverAgent#705]
## The problem

While executing the following command in landscape, appium treats 'x' as 'y' and vise versa.

## **Environment**

Appium version (or git revision) that exhibits the issue: Appium v1.7.0-beta, v1.6.6-beta.4
Last Appium version that did not exhibit the issue (if applicable): Appium v1.6.3
Desktop OS/version used to run Appium: Sierra Version 10.12.5
Mobile platform/version under test: iPad Air 2, iOS 11.0
Real device or emulator/simulator: simulator

## **Details**

Also by executing the commands added, element is moved from the bottom of the screen by the diagonal (not just by one of the axis)

## **Appium logs**

Found matching gesture: drag

[debug] [JSONWP Proxy] Proxying [POST /wda/dragfromtoforduration] to [POST http://localhost:8100/session/1EB033B5-BDAD-4132-950A-D1B91D8C0437/wda/dragfromtoforduration] with body: {"fromX":716,"fromY":209,"toX":1543,"toY":418,"duration":1}

## **Code To Reproduce Issue**

new TouchAction(appiumDriver).press(716, 209).waitAction(1000).moveTo(827, 209).release().perform();

**Comment by marekcirkos** [comment]
@mykola-mokhnach Is this Appium issue or your coordinate swapping?

**Comment by mykola-mokhnach** [comment]
No, it started to happen after Xcode update to version 9. I've added a [PR](https://github.com/facebook/WebDriverAgent/pull/714) a while ago to fix that and this PR has been even already merged into our local forked WDA.

**Comment by marekcirkos** [comment]
Merged. @tmytlovych Can you confirm it works for you?

**Comment by quannd307** [comment]
Hi @mykola-mokhnach , about this issue [https://github.com/appium/appium/issues/9329](url), I've checked out the latest release of WDA (v2.58.1) but this issue still happens. Can you confirm that issue is solved?
Thank in advance!

**Comment by marekcirkos** [comment]
@mykola-mokhnach Is it https://github.com/facebook/WebDriverAgent/issues/856 related?

**Comment by mykola-mokhnach** [comment]
All of them are related if we are talking about non-default orientation.
